### PR TITLE
feat: pagination to 25 and allow disable_pagination

### PR DESCRIPTION
**Pull request by leangseu-edx**
ticket: https://2u-internal.atlassian.net/browse/AU-1024

What change:
- page size default to 25
- if the url include `disable_pagination=1`, it will disable pagination

### CODE REVIEW

File: src/containers/CourseList/hooks.test.js [hunk 1]
```diff
@@ -56,6 +62,17 @@ describe('CourseList hooks', () => {
           pageSize: ListPageSize,
         });
       });
+      it('loads current course list with page size 0 there is query disable_pagination=1', () => {
```
Comment (jansenk) [inline src/containers/CourseList/hooks.test.js:65]: loads current course list with page size 0 **if/when** there is query **param** disable_pagination=1'

File: src/containers/CourseList/NoCoursesView/index.test.jsx [hunk 2]
```diff
@@ -3,20 +3,14 @@ import { shallow } from 'enzyme';
...
+jest.mock('hooks', () => ({
+  reduxHooks: {
+    usePlatformSettingsData: jest.fn(() => ({
       courseSearchUrl: 'course-search-url',
-    }),
+    })),
   },
 }));
 
-jest.mock('containers/Dashboard/hooks', () => ({
```
Comment (jansenk) [inline src/containers/CourseList/NoCoursesView/index.test.jsx:16]: Why was this removed?

File: src/containers/CourseList/NoCoursesView/index.test.jsx [hunk 3]
```diff
@@ -3,20 +3,14 @@ import { shallow } from 'enzyme';
 
 import EmptyCourse from '.';
 
-jest.mock('data/redux', () => ({
-  hooks: {
-    useRecommendedCoursesData: jest.fn(() => ({ courses: [], isControl: false })),
```
Comment (jansenk) [inline src/containers/CourseList/NoCoursesView/index.test.jsx:8]: Why was this changed?

File: src/widgets/RecommendationsPanel/LoadedView.jsx [hunk 4]
```diff
@@ -48,14 +48,18 @@ export const LoadedView = ({
...
+};
+
 LoadedView.propTypes = {
   courses: PropTypes.arrayOf(PropTypes.shape({
     courseKey: PropTypes.string,
     title: PropTypes.string,
     logoImageUrl: PropTypes.string,
     marketingUrl: PropTypes.string,
   })).isRequired,
-  isControl: PropTypes.oneOf([true, false, null]).isRequired,
```
Comment (jansenk) [inline src/widgets/RecommendationsPanel/LoadedView.jsx:58]: This change seems unrelated?

**Review by jansenk — DISMISSED** [review]
There are a few changes that seem unrelated in here? Maybe I just don't understand the code.

Other than that, one wording suggestion, other than that, looks good.

File: src/containers/CourseList/NoCoursesView/index.test.jsx (hunk 2 above)
Comment (leangseu-edx) [reply src/containers/CourseList/NoCoursesView/index.test.jsx]: Sorry it was left over for test fix. I will post a different pr for that.
